libs/header/p2p/server.go

```diff
@@ -201,7 +201,13 @@ func (serv *ExchangeServer[H]) handleRequest(from, to uint64) ([]H, error) {
 	}
 
 	log.Debugw("server: handling headers request", "from", from, "to", to)
-	headersByRange, err := serv.getter.GetRangeByHeight(ctx, from, to)
+	if !serv.store.HasAt(to - 1) {
+		span.SetStatus(codes.Error, header.ErrNotFound.Error())
+		log.Warnw("server: requested headers not stored")
```

Comment:
I'd do a debug log as this can be more frequent - also log the `to` height

Reply:
Changed to debug. But I do not think we really need to print `to` height one more time. We are logging 
`log.Debugw("server: handling headers request", "from", from, "to", to)`  before `HasAt` check.

Reply:
those can always be lost though, meaning logs get cloudy so it's nice to know which request this is attributed to

Reply:
ok, np.